### PR TITLE
feat(mlop-2456): add protection to host setting on cassandra_client

### PR DESCRIPTION
**Pull request by albjoaov**
# What 🔧 

- Makefile: changes to exclude `venv` files from checks and formatting
- Add `_validate_and_format_cassandra_host` and unit tests 
- `tests/unit/butterfree/transform/transformations/conftest.py`: fixing a test

### CODE REVIEW

File: butterfree/clients/cassandra_client.py [hunk 1]
```diff
@@ -53,12 +59,48 @@ def __init__(
...
+
+        Args:
+            host (str | list): The Cassandra host input, which can be a comma-separated
+                            string or a list of string endpoints.
+
+        Returns:
+            list: A list of formatted Cassandra host strings.
+
+        Raises:
+            ValueError: If the input is an empty string or if it is not a string
```
Comment (marcoVinha) [inline butterfree/clients/cassandra_client.py:87]:
```suggestion
            ValueError: If the input is an empty list or if it is not a string
```

File: butterfree/clients/cassandra_client.py [hunk 2]
```diff
@@ -16,6 +16,12 @@
 from typing_extensions import TypedDict
 
 from butterfree.clients import AbstractClient
+from butterfree.configs.logger import __logger
+
+logger = __logger("cassandra_client")
+
+EMPTY_STRING_HOST_ERROR = "The value of Cassandra host is empty. Please fill correctly with your endpoints"  # noqa: E501
```
Comment (marcoVinha) [inline butterfree/clients/cassandra_client.py:23]: If we're importing this strings elsewhere in our code, isn't it better to store them in a `cassandra_client_errors.py` file, in an Enum, or something like that?

File: tests/unit/butterfree/transform/transformations/conftest.py [hunk 3]
```diff
@@ -62,7 +62,7 @@ def target_df_spark(spark_context, spark_session):
             "timestamp": "2016-04-11 11:31:11",
             "feature1": 200,
             "feature2": 200,
-            "feature__cos": 0.48718767500700594,
+            "feature__cos": 0.4871876750070059,
```
Comment (marcoVinha) [inline tests/unit/butterfree/transform/transformations/conftest.py:65]: Why this change?

Comment (albjoaov) [reply tests/unit/butterfree/transform/transformations/conftest.py:65]: The expected value it is without this last number

File: butterfree/clients/cassandra_client.py (hunk 2 above)
Comment (albjoaov) [reply butterfree/clients/cassandra_client.py:23]: I don't see this need. It's just used there. If something changes we can create something more generic

File: butterfree/clients/cassandra_client.py (hunk 1 above)
Comment (albjoaov) [reply butterfree/clients/cassandra_client.py:87]: Actually it's both. Empty string or list is the problem

File: butterfree/clients/cassandra_client.py (hunk 2 above)
Comment (marcoVinha) [reply butterfree/clients/cassandra_client.py:23]: Oh, I see! Got confused thinking it was used somewhere else :)

**Review by marcoVinha — APPROVED** [review]

File: butterfree/clients/cassandra_client.py (hunk 1 above)
Comment (albjoaov) [reply butterfree/clients/cassandra_client.py:87]: Updated
